src/audio/iir.c

```diff
@@ -89,44 +94,45 @@ int32_t iir_df2t(struct iir_state_df2t *iir, int32_t x)
 			 * Q2.30 x Q1.31 -> Q3.61
 			 * Shift Q3.61 to Q3.31 with rounding
 			 */
-			acc = ((int64_t) iir->coef[c + 4]) * in + iir->delay[d];
-			tmp = (int32_t) Q_SHIFT_RND(acc, 61, 31);
+			acc = ((int64_t)iir->coef[c + 4]) * in + iir->delay[d];
+			tmp = (int32_t)Q_SHIFT_RND(acc, 61, 31);
 
 			/* Compute 1st delay */
 			acc = iir->delay[d + 1];
-			acc += ((int64_t) iir->coef[c + 3]) * in; /* Coef  b1 */
-			acc += ((int64_t) iir->coef[c + 1]) * tmp; /* Coef a1 */
+			acc += ((int64_t)iir->coef[c + 3]) * in; /* Coef  b1 */
+			acc += ((int64_t)iir->coef[c + 1]) * tmp; /* Coef a1 */
 			iir->delay[d] = acc;
 
 			/* Compute 2nd delay */
-			acc = ((int64_t) iir->coef[c + 2]) * in; /* Coef  b2 */
-			acc += ((int64_t) iir->coef[c]) * tmp; /* Coef a2 */
+			acc = ((int64_t)iir->coef[c + 2]) * in; /* Coef  b2 */
+			acc += ((int64_t)iir->coef[c]) * tmp; /* Coef a2 */
 			iir->delay[d + 1] = acc;
 
 			/* Gain, output shift, prepare for next biquad
 			 * Q2.14 x Q1.31 -> Q3.45, shift too Q3.31 and saturate
 			 */
-			acc = ((int64_t) iir->coef[c + 6]) * tmp; /* Gain */
+			acc = ((int64_t)iir->coef[c + 6]) * tmp; /* Gain */
 			acc = Q_SHIFT_RND(acc, 45 + iir->coef[c + 5], 31);
```

Comment:
would be good to remove the magic numbers here...

Reply:
The c + 6  is the per biquad gain parameter as commented. The numbers in Q_SHIFT_RND are not fully covered by comments though. The shift is done to convert from Q45 to Q31 and apply additional shift from c + 5 shift parameter. I'll add that and push new version of PR... It will also test if 0day test could complete this the the check.

I will later try to rewrite this by using the the new structs in uapi/eq.h. Doing biquad->gain might be faster than this iir->coef[c+6].

Reply:
@singalsu ok, we need to get rid of those magic numbers though, incremental patch ? Btw, did you see any errors when building on APL. I'm not sure what 0day is complaining about.